src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/Analizador.kt

```diff
@@ -0,0 +1,48 @@
+package ar.edu.unahur.obj2.servidorWeb
+
+import java.time.LocalDateTime
+
+class Pedido(val modulo : Modulo?,val respuesta: Respuesta, val ip : String, val url : String, val fecha : LocalDateTime){}
+
+abstract class Analizador(){
+    val pedidos = mutableListOf<Pedido>()
+
+    open fun agregarPedido(pedido : Pedido) : Unit {
+        pedidos.add(pedido)
+    }
+
+    fun respuestas() = pedidos.map{it.respuesta}
+}
+
+class AnalizadorDemoras(val tiempoMinimo: Int) : Analizador(){
+
+    fun cantidadDeDemoras() = pedidos.filter{ it.respuesta.tiempo > tiempoMinimo }.size
+}
+
+class AnalizadorIPs(val ipsSospechosas : Set<String>) : Analizador() {
+
+    override fun agregarPedido(pedido : Pedido) : Unit {
+        if(ipsSospechosas.contains(pedido.ip)){
+            pedidos.add(pedido)
+        }
+    }
+
+    fun cantidadDePedidos(ipSospechosa : String) = pedidos.filter{ it.ip == ipSospechosa}.count()
+
+    fun moduloMasConsultado() = pedidos.groupBy{it.modulo}.maxBy{it.component2().size}?.component1()
+
+    fun ipsSospechosasConRuta(ruta : String) = pedidos.filter{ it.url == ruta}.map{it.ip}.toSet()
+
+}
+
+class AnalizadorEstadisticas() : Analizador(){
+
+    fun tiempoPromedio() = respuestas().sumBy{it.tiempo} / respuestas().count()
```

Comment:
Está muy bien. Otra forma de hacerlo:
```suggestion
    fun tiempoPromedio() = respuestas().map { it.tiempo }.average()
```